presto-main/src/main/java/com/facebook/presto/cost/ExchangeStatsRule.java

```diff
@@ -62,6 +74,11 @@ public Pattern getPattern()
             }
         }
 
+        if (exchangeNode.getType() == REPLICATE) {
+            verify(exchangeNode.getScope() == REMOTE, "unexpected scope 'LOCAL' for exchange type 'REPLICATE'");
+            estimate.map(stats -> stats.mapOutputRowCount(rowsCount -> rowsCount * numberOfNodes.getAsInt()));
```

Comment:
I am not sure if it is a correct approach. Physically rows are replicated, but logically the number of rows in a relation remains the same. 

If you multiply rows here, then you would need to divide appropriately them in `JoinStatsRule`, otherwise you will get wrong stats estimation. That is why I do not think we should go this way. I would suggest you to multiply cost instead.